src/Tools/Source/CompilerGeneratorTools/Source/CSharpErrorFactsGenerator/ErrorGenerator.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// We only build the Source Generator in the netstandard target
+#if NETSTANDARD
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Internal.CSharpErrorFactsGenerator
+{
+    [Generator]
+    public sealed partial class ErrorGenerator : IIncrementalGenerator
+    {
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            var provider = context.SyntaxProvider.CreateSyntaxProvider(
+                predicate: (n, _) => n is EnumDeclarationSyntax enumDeclaration && enumDeclaration.Identifier.ValueText.Equals("ErrorCode", StringComparison.Ordinal),
```

Comment:
predicate seems a bit too broad.  like... if anyone else makes an enum with this name, then we generate this :)

(or is there something ensuring that thsi would only run inside the compilers, where we are likely only to see thsi once)?

Reply:
@CyrusNajmabadi The generator is only referenced inside compilers. I don't think there will be another `ErrorCode` enum in the C# compiler.